libgnucash/engine/gnc-uri-utils.c

```diff
@@ -342,3 +342,81 @@ gchar *gnc_uri_add_extension ( const gchar *uri, const gchar *extension )
     /* Ok, all tests passed, let's add the extension */
     return g_strconcat( uri, extension, NULL );
 }
+
+gboolean gnc_uri_absolute_associate_file_path (const gchar *uri,
+                                               const gchar *path_head,
+                                               gchar **ret_uri)
+{
```

Comment:
Related question: can the base path also be a location ? Your code so far only accepts a file uri as base path. However perhaps all files are on an ftp server instead. That's currently not supported and perhaps for now we should keep it that way. If we do want to support it the relative path handling may require two functions: one for local file system paths (via gnc-filepath-utils) and one for other uri based base paths.

Reply:
Do you agree with the following, just trying to make sure I understand the starting point.
The path head will be stored in a form like...
Windows

- file://localhost/c:/somedir
- file:///c:/somedir
- file:///c:/ 

Linux

- file://locallhost/usr/somedid
- file:///usr/somedir
- file:///

A Location Association must have a valid scheme/protocol
A File Association could be relative or absolute
The stored associate value will be in the form depending on OS and whether file or location based...

- file:///c:/User/Admin/some-file.txt
- some-file.txt
- docs/list/some-file.txt
- http://www.gnucash.org
- ftp://myname@host.dom/home/myname/some-file.txt
- file:///usr/somedir/some-file.txt

This would make invalid scheme/protocols a relative path and can be made absolute by adding to path head and with a '/' depending on whether the path head ended with a '/'